core-java-modules/core-java-numbers-5/src/main/java/com/baeldung/uniquerng/UniqueRng.java

```diff
@@ -0,0 +1,42 @@
+package com.baeldung.uniquerng;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+public class UniqueRng implements Iterator<Integer> {
+
+    private int size;
+    private List<Integer> numbers = new ArrayList<>();
+
+    public UniqueRng(int size, boolean zeroBased) {
+        this.size = size;
+        int start = (zeroBased ? 0 : 1);
+        int limit = (zeroBased ? size - 1 : size);
+
+        for (int i = start; i <= limit; i++) {
+            numbers.add(i);
+        }
+
+        Collections.shuffle(numbers);
+    }
+
+    @Override
+    public Integer next() {
+        if (numbers.size() < 1)
```

Comment:
Using isEmpty is easier to read and sometimes more performant, so do you mind changing to `numbers.isEmpty();`